nms/app/packages/magmalte/app/components/context/RefreshContext.js

```diff
@@ -0,0 +1,305 @@
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb informations', {
+        variant: 'error',
+      });
+      return {
+        ...refreshingState,
+        state: {
+          ...refreshingState.state,
+          enbInfo: {
+            ...refreshingState.state.enbInfo,
+            [id]: {enb_state: {}, enb: newEnb},
+          },
+        },
+      };
+    }
+  } else {
+    let enb = {};
+    try {
+      enb = await MagmaV1API.getLteByNetworkIdEnodebs({networkId});
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb information', {
+        variant: 'error',
+      });
+    }
+
+    if (!enb) {
+      return;
+    }
+    const requests = Object.keys(enb).map(async k => {
+      try {
+        const {serial} = enb[k];
+        // eslint-disable-next-line max-len
+        const enbSt = await MagmaV1API.getLteByNetworkIdEnodebsByEnodebSerialState(
+          {
+            networkId: networkId,
+            enodebSerial: serial,
+          },
+        );
+        return [enb[k], enbSt ?? {}];
+      } catch (e) {
+        enqueueSnackbar?.('failed fetching enodeb information', {
+          variant: 'error',
+        });
+        return [enb[k], {}];
+      }
+    });
+    try {
+      const enbResp = await Promise.all(requests);
+      const enbInfo = {};
+      if (enbResp) {
+        enbResp.filter(Boolean).forEach(r => {
+          if (r.length > 0) {
+            const [enb, enbSt] = r;
+            if (enb != null && enbSt != null) {
+              enbInfo[enb.serial] = {
+                enb: enb,
+                enb_state: enbSt,
+              };
+            }
+          }
+        });
+        return {...refreshingState, state: {enbInfo: enbInfo}};
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb information', {
+        variant: 'error',
+      });
+    }
+  }
+}
+
+async function fetchGateways(props: fetchProps) {
+  const {networkId, refreshingState, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== null) {
+    const gwId = id;
+    try {
+      const gateway = await MagmaV1API.getLteByNetworkIdGatewaysByGatewayId({
+        networkId: networkId,
+        gatewayId: gwId,
+      });
+      if (gateway) {
+        const newLteGateways = {
+          ...refreshingState.state,
+          [gwId]: gateway,
+        };
+        return {...refreshingState, state: newLteGateways};
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching gateway informations', {
+        variant: 'error',
+      });
+    }
+  } else {
+    try {
+      const gateways = await MagmaV1API.getLteByNetworkIdGateways({
+        networkId,
+      });
+      if (gateways) {
+        return {
+          ...refreshingState,
+          state: gateways,
+        };
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching gateway informations', {
+        variant: 'error',
+      });
+    }
+  }
+}
+async function fetchSubscribers(props: fetchProps) {
```

Comment:
This might be missing the logic currently present in InitSubscriberState, As mentioned earlier, we should probably merge the init and the fetch methods. https://github.com/magma/magma/blob/master/nms/app/packages/magmalte/app/state/lte/SubscriberState.js#L35